Code/src/main/java/algorythm/BFS.java

```diff
@@ -0,0 +1,29 @@
+package algorythm;
+
+import graph.Node;
+
+import java.util.*;
+
+public class BFS<T> implements Iterator<Node<T>> {
+    private Queue<Node> queue = new ArrayDeque<>();
+    private List<Node> visited = new ArrayList<>();
+    public BFS(Node<T> startingNode) {
+        queue.add(startingNode);
+    }
+
+    @Override
+    public boolean hasNext() {
+        return !queue.isEmpty();
+    }
+
+    @Override
+    public Node<T> next() {
+        Node<T> currentNode = queue.remove();
+        for(Node<T> nodeToAdd: currentNode.getNeighbours()){
+            if(!visited.contains(nodeToAdd))
+                queue.add(nodeToAdd);
+        }
+        visited.add(currentNode);
```

Comment:
Ta linijka mogłaby być właściwie przed pętlą dodającą węzły do kolejki do odwiedzenia, ale chyba nie ma to większego znaczenia.

Reply:
Ta linijka, to mam na myśli 26